packages/web/src/scss/tools/_form-fields.scss

```diff
@@ -0,0 +1,136 @@
+    content: '*';
+    color: form-fields-theme.$label-required-color;
+}
+
+@mixin input($fieldType) {
+    @if $fieldType == 'inline' {
+        appearance: none;
+        margin: form-fields-theme.$inline-field-input-margin;
+        color: form-fields-theme.$inline-field-input-color-unchecked;
+        border: form-fields-theme.$inline-field-input-color-default solid currentcolor;
+
+        &:focus-visible {
+            outline: 0;
+            box-shadow: form-fields-theme.$input-focus-shadow;
+        }
+    }
+
+    @if $fieldType == 'box' {
+        @include typography.generate(form-fields-theme.$input-typography);
+
+        display: block;
+        width: 100%;
+        padding: form-fields-theme.$box-field-input-padding-y form-fields-theme.$box-field-input-padding-x;
+        color: form-fields-theme.$box-field-input-color-default;
+        border: form-fields-theme.$box-field-input-border-width
+            form-fields-theme.$box-field-input-border-style
+            form-fields-theme.$box-field-input-border-color-default;
+        border-radius: form-fields-theme.$input-border-radius;
+        background: form-fields-theme.$box-field-input-background;
+
+        &::placeholder {
+            color: form-fields-theme.$box-field-input-placeholder-color-default;
+            opacity: 1;
+        }
+    }
+}
+
+@mixin input-disabled() {
+    color: form-fields-theme.$input-color-disabled;
+}
+
+@mixin message() {
+    @include typography.generate(form-fields-theme.$message-typography);
+
+    display: block;
+    margin-top: form-fields-theme.$message-margin-top;
+    color: form-fields-theme.$message-color-default;
+}
+
+@mixin message-disabled() {
+    color: form-fields-theme.$message-color-disabled;
+}
+
+@mixin box-field-variants($fieldName) {
+    @each $variant-name, $variant-value in form-fields-settings.$box-field-variants {
+        :is(.#{$fieldName}--#{$variant-name}, .#{$fieldName}.has-#{$variant-name}) > .#{$fieldName}__input {
+            border-color: map.get($variant-value, input-border-color);
+        }
+
+        @if $fieldName == 'TextField' {
+            :is(.#{$fieldName}--#{$variant-name}, .#{$fieldName}.has-#{$variant-name})
+                .#{$fieldName}__passwordToggle
+                > .#{$fieldName}__input
+                ~ .#{$fieldName}__passwordToggle__button::before {
+                border-color: map.get($variant-value, input-border-color);
+            }
+        }
+
+        :is(.#{$fieldName}--#{$variant-name}, .#{$fieldName}.has-#{$variant-name})
+            > :is(.#{$fieldName}__message, [data-element='validator_message']) {
+            color: map.get($variant-value, message-color);
+        }
+    }
+}
+
+@mixin box-field-fluid() {
+    width: 100%;
+}
+
+@mixin box-field-focus-visible() {
+    border-color: form-fields-theme.$box-field-input-border-color-focus;
+    outline: 0;
+    box-shadow: form-fields-theme.$input-focus-shadow;
+}
+
+@mixin box-field-disabled-input() {
```

Comment:
maybe the disabled textarea should block resize? But I am not sure, what is better UX.

Reply:
Good question! Let's UX designer decide it. 🙂